PubnativeLite/PubnativeLite/Ad Cache/HyBidVideoAdCache.m

```diff
@@ -0,0 +1,51 @@
+//
+//  Copyright © 2020 PubNative. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "HyBidVideoAdCache.h"
+
+@implementation HyBidVideoAdCache
+
+- (void)dealloc {
+    self.videoAdCache = nil;
+}
+
++ (instancetype)sharedInstance {
+    static HyBidVideoAdCache *_sharedInstance;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedInstance = [[HyBidVideoAdCache alloc] init];
+        _sharedInstance.videoAdCache = [[NSMutableDictionary alloc] init];
```

Comment:
I think it's better to initialize videoAdCache in init method.

Reply:
Why @faresbh ? Since we have an another class called `HyBidAdCache` initialized with a similar singleton object, i thought it'd be okay to follow the same format.

Reply:
It's cleaner to do so. Imagine we have more init stuff to do, then we know that everything will be handed to init and the shared instance will be obviously that init. It does not harm in this case here ;)

Reply:
Cool 👍 Since it does not harm in this case, i'll leave it like this. But in the future if we have a scenario like you've said, then it'd be better to use an `init` for sure 🤩